core-plugins/src/main/java/io/cdap/plugin/batch/aggregator/function/NumberSelection.java

```diff
@@ -19,172 +19,57 @@
 import io.cdap.cdap.api.data.format.StructuredRecord;
 import io.cdap.cdap.api.data.schema.Schema;
 
-import java.util.List;
-import javax.annotation.Nullable;
-
 /**
  * Base class for number based selection functions.
  * Allows subclasses to implement typed methods instead of implementing their own casting logic.
  * Guarantees that only methods for one type will be called for each aggregate. For example,
- * if {@link #operateOn(StructuredRecord)} is called, only {@link #operateOnInt(int, StructuredRecord)} will be called.
+ * if {@link #select(StructuredRecord, StructuredRecord)} is called,
+ * only {@link #combineInt(int, int)} will be called.
  */
 public abstract class NumberSelection implements SelectionFunction {
-  private final SelectionFunction delegate;
   private final String fieldName;
+  private final Schema.Type fieldType;
 
-  public NumberSelection(final String fieldName, @Nullable Schema fieldSchema) {
+  public NumberSelection(String fieldName, Schema fieldSchema) {
     this.fieldName = fieldName;
-    // if schema is not known before we start getting records, just use doubles.
```

Comment:
there used to be null handling before, how come it is being removed? Was that actually never supported?

Reply:
By default it uses double to handle null, but a null schema scenario actually doesn't exist. If a field does not exist, the validation should fail. If a field does exist, then it has to have a field schema. Previously this was mainly used to validate and unit test that a null schema can fetch the result, so I removed this.